Xamarin.Forms.Xaml/MarkupExtensions/OnIdiomExtension.cs

```diff
@@ -0,0 +1,78 @@
+﻿using System;
+using System.Globalization;
+using System.Reflection;
+
+namespace Xamarin.Forms.Xaml
+{
+	[ContentProperty("Default")]
+	public class OnIdiomExtension : IMarkupExtension
+	{
+		// See Device.Idiom
+
+		public object Default { get; set; }
+		public object Unsupported { get; set; }
+		public object Phone { get; set; }
+		public object Tablet { get; set; }
+		public object Desktop { get; set; }
+		public object TV { get; set; }
+		public object Watch { get; set; }
+
+		public IValueConverter Converter { get; set; }
+
+		public object ConverterParameter { get; set; }
+
+		public object ProvideValue(IServiceProvider serviceProvider)
+		{
+			var lineInfo = (serviceProvider.GetService(typeof(IXmlLineInfoProvider)) as IXmlLineInfoProvider)?.XmlLineInfo;
```

Comment:
I recently added a generic `GetService<T>` you might want to use.

also, do `serviceProvider?.GetService...`